atoMEC/mathtools.py

```diff
@@ -339,23 +338,38 @@ def chem_pot(orbs):
     if config.unbound == "quantum":
         for i in range(config.spindims):
             if config.nele[i] != 0:
-                soln = optimize.root_scalar(
-                    f_root_qu,
-                    x0=mu0[i],
-                    args=(
-                        orbs.eigvals[:, i],
-                        orbs.occ_weight[:, i],
-                        config.nele[i],
-                    ),
-                    method="brentq",
-                    bracket=[-100, 100],
-                    options={"maxiter": 100},
-                )
-                mu[i] = soln.root
+                if config.mu[i] == 0:
```

Comment:
Please bring this if statement 'to the front', i.e. outside the other if statements. Then no repeated code.

Also please make same changes to unbound="ideal"